docs/rules/assertions.md

```diff
@@ -27,9 +27,12 @@ assert/my-assertion-name | [Assertion object](#assertion-object) | An assertion
 
 Property | Type | Description
 -- | -- | --
-subject | `string` \| [`string`] | **REQUIRED.** The [OpenAPI node type](#openapi-node-types) that the lint evaluates. Use with `context` for more control.
+subject | `string` \| [`string`] | **REQUIRED.** Locates the [OpenAPI node type](#openapi-node-types) that the lint evaluates. Use with `if` for more control.
 property | `string` \| [`string`] \| null | Property name corresponding to the [OpenAPI node type](#openapi-node-types). If a list of properties is provided, assertions will evaluate against each property in the sequence. If not provided (or null), assertions will evaluate against the key names for the subject node type. See [property example](#property-example).
-context | [Context object](#context-object) | The context influences evaluation for assertions. When `matchParentKeys` or `excludeParentKeys` is used in the `context` object, it evaluates the specified subset of the subject type. The resolution of reference objects is done at the context level. If no context is provided, it evaluates the assertion for all instances of the given type. See [context example](#context-example).
+filterInParentKeys | [`string`] | The name of the subject's parent key that locates where assertions run. An example value given the subject `Operation` could be `filterInParentKeys: [get, put]` means that only `GET` and `PUT` operations are evaluated for the assertions.
+filterOutParentKeys | [`string`] | The name of the subject's parent key that excludes where assertions run. An example value given the subject `Operation` could be `filterOutParentKeys: [delete]` means that all operations except `DELETE` operations are evaluated for the assertions.
+matchParentKeys | `string` | Applies a regex pattern to the subject's parent keys to determine where assertions run. An example value given the subject `Operation` could be `matchParentKeys: /^p/` means that `POST`, `PUT`, and `PATCH` operations are evaluated for the assertions.
+if | [`Assertion` object] | The `if` list influences evaluation for assertions by only evaluating assertions when the `if` arguments evaluate to true. The `if` argument must be structured like a valid assertion with the exception of no nested `if` values. When using the `if` list, it evaluates the locator specified with any assertions, and then continues to evaluate based on that subset of the values. The resolution of reference objects is done at the `if` level. The `if` list is evaluated from top to bottom in that order. If no `if` list is provided, it evaluates the assertion for all instances of the given type. See [if example](#if-example). The `if` evaluation does not result in any problems (it is used for filtering where to run assertions); therefore, the `message`, `suggest`, and `severity` properties are ignored.
```

Comment:
We should state somewhere that the order of the array matters: it should go from the `Root` down to the subject (including).